### DISCUSSION TITLE
App upload to Github Pages

### DISCUSSION THREAD

**Discussion by lukbar48** [mswjs/msw#997] [Q&A] [answered]
Hi, I created my react app with msw and I uploaded it on github pages. When the data should be fetched I get 405 error, however it's working on localhost. I'm not sure if it's my fault or it's just not possible to host app with msw.
Here is the app and msw config code:
https://github.com/lukbar48/dietmaster/blob/main/src/mocks/handlers/patients.js
https://lukbar48.github.io/dietmaster/

**Comment by kettanaito** [comment, marked as answer]
Hi, @lukbar48. 

You get 405 from GitHub because POST requests are forbidden against GitHub pages. 

When you open the browser's console and refresh the page, there's no `[MSW] Mocking enabled` message displayed. That's a straight sign that MSW is not included in your application. If you go to your `src/index.tsx`, you can see the exact [condition that excludes it](https://github.com/lukbar48/dietmaster/blob/3ad3b8557f6240a84630f2f362786deefb2eb21f/src/index.tsx#L7). MSW is, generally, a development tool, that's why most resources online recommend including it only when developing. If I understand your use case correctly, you wish to have MSW as a server replacement for your application _in production_. In that case, remove the `if` condition that checks `process.env.NODE_ENV` and the library will be included in your production build.
